app/bundles/ReportBundle/Controller/ReportController.php

```diff
@@ -783,14 +783,42 @@ public function exportAction($objectId, $format = 'csv')
         $fromDate = $session->get('mautic.report.date.from', (new \DateTime('-30 days'))->format('Y-m-d'));
         $toDate   = $session->get('mautic.report.date.to', (new \DateTime())->format('Y-m-d'));
 
-        $dynamicFilters = $session->get('mautic.report.'.$objectId.'.filters', []);
+        $date = (new DateTimeHelper())->toLocalString();
+        $name = str_replace(' ', '_', $date) . '_' . InputHelper::alphanum($entity->getName(), false, '-');
+        $options = array( 'dateFrom' => new \DateTime($fromDate),'dateTo' => new \DateTime($toDate));
+
+        if($format === 'csv') {
+            $response = new HttpFoundation\StreamedResponse(
+                function () use ($model, $fromDate, $toDate, $entity, $format, $name, $options) {
+                    $options['paginate'] = true;
+                    $options['ignoreGraphData'] = true;
+                    $options['limit'] =
+                    $reportData['totalResults'] = 10000;
+                    $options['page'] = 1;
+                    $handle = fopen('php://output', 'r+');
+                    while ($reportData['totalResults'] >= ($options['page'] - 1) *  $options['limit']) {
+                        $reportData = $model->getReportData($entity, null, $options);
+                        $model->exportResults($format, $entity, $reportData, $handle, null, $options['page']);
+                        $options['page']++;
+                    }
+                    fclose($handle);
+                }
+            );
 
-        $reportData = $model->getReportData($entity, null, [
-            'dateFrom'       => new \DateTime($fromDate),
-            'dateTo'         => new \DateTime($toDate),
-            'dynamicFilters' => $dynamicFilters,
-        ]);
+            $response->headers->set('Content-Type', 'application/force-download');
+            $response->headers->set('Content-Type', 'application/octet-stream');
+            $response->headers->set('Content-Disposition', 'attachment; filename="' . $name . '.'.$format.'"');
+            $response->headers->set('Expires', 0);
+            $response->headers->set('Cache-Control', 'must-revalidate');
+            $response->headers->set('Pragma', 'public');
+        }else{
```

Comment:
Please check the Mautic coding standard and the pre-commit hook:

https://github.com/mautic/mautic/blob/staging/.github/CONTRIBUTING.md#coding-standards

There are also other places where your code is not following the standard.